.github/workflows/base-image.yml

```diff
@@ -45,7 +45,7 @@ jobs:
             *.cache-to=type=gha,mode=max,scope=${{ matrix.image.name }}
           push: ${{ github.event_name == 'push' && vars.PUSH_TO_REGISTRY == 'true' && github.ref_type == 'tag' }}
       - name: Point IOC Dockerfile to image that was just built
-        run: sed -i "s/v.*-dev/$TAG/" Dockerfile
+        run: sed -i -E "s/(v[0-9]+\.[0-9]+\.[0-9]+-dev)/$TAG/" Dockerfile
```

Comment:
Nit: we don't need the group matching here, that is, you can drop the parentheses around the regexp.